lib/pure/views.nim

```diff
@@ -0,0 +1,148 @@
+## View is a type representing a range of elements in an array. In can be thought as a pointer plus a size.
+## View can be created to an arbitrary memory segment and can additionally keep a single ``ref`` object alive.
+##
+## This module defines views and several helper operations on them. All functions in this module (except for unsafe version of ``initView``) are designed to be memory safe (e.g. they raise exception on out-of-bounds accesses).
+import strutils
+
+const
+  needGcKeep = not (compileOption("gc", "boehm") or compileOption("gc", "none"))
+
+type
+  View*[T] = object
+    data: ptr T
+    length: int
+    when needGcKeep:
+      gcKeep: RootRef
```

Comment:
`needGcKeep` is defined as `not (compileOption("gc", "boehm") or compileOption("gc", "none"))`, so it can't be disabled even globally.

Reply:
yes that is what I mean. It can be disabled/enabled globally. But it should be possible to enable/disable it locally. Meaning a different type that has the RootRef and one that does not. Similar to ptr/ref being a different type. One does keep the object alive the other one doesn't.